daemon/config/config_linux.go

```diff
@@ -245,6 +248,30 @@ func validatePlatformConfig(conf *Config) error {
 	return verifyDefaultCgroupNsMode(conf.CgroupNamespaceMode)
 }
 
+// verifyExecOptions checks if the configured exec-opts are valid for the platform.
+// It returns an error if the exec-options are formatted incorrectly, or when
+// options are used that are not supported on this platform.
+func verifyExecOptions(execOptions []string) error {
+	for _, opt := range execOptions {
+		k, v, ok := strings.Cut(opt, "=")
+		k = strings.ToLower(strings.TrimSpace(k))
+		v = strings.TrimSpace(v)
+		if !ok || k == "" || v == "" {
+			return fmt.Errorf("invalid exec-opt (%s): must be formatted 'opt=value'", opt)
+		}
```

Comment:
Looks like this part is duplicated in all platform specific `verifyExecOptions` implementations.

I think we can keep the loop separate and only make a platform-specific `func validateExecOption(key, value string) error` function that validates a single option.

Reply:
Yeah, this one was a bit tricky (and I so much would love to get rid of  some of  the "too permissive" handling).

I went a bit back-and-forth on this one; initially I had a non-platform-specific validate step that only validated the format, but then I still had to do similar steps in the per-platform one (trimming, lowercasing, all the things).

I then tried a single validate function but some conditionals (`if runtime.GOOS != "windows" { return err ..}`), which was "ok(ish)" for the set of options we currently had, but I wasn't sure if that was the cleanest approach. So I somewhat  considered the  duplication to  be OK (keeping the `_linux.go` and `_windows.go` similar, to allow doing a side-by-side  diff).

I was somewhat trying to avoid differentiating in too many places because we had the `validatePlatformConfig` as place  to have things  that are platform-specific.

I can have another look to see if there's other options (I may move the first couple of commits separate to get those in already)